phoenix-core-client/src/main/java/org/apache/phoenix/expression/function/DecodeBinaryFunction.java

```diff
@@ -0,0 +1,128 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.phoenix.expression.function;
+
+import java.sql.SQLException;
+import java.util.Base64;
+import java.util.List;
+
+import org.apache.hadoop.hbase.io.ImmutableBytesWritable;
+import org.apache.hadoop.hbase.util.Bytes;
+import org.apache.phoenix.expression.Expression;
+import org.apache.phoenix.parse.FunctionParseNode;
+import org.apache.phoenix.schema.IllegalDataException;
+import org.apache.phoenix.schema.types.PDataType;
+import org.apache.phoenix.schema.tuple.Tuple;
+import org.apache.phoenix.schema.types.PVarchar;
+
+/**
+ * Decodes binary data in various formats (Hex, Base64, HBase) by extending the DecodeFunction.
+ * <p>
+ * Usage:
+ * DECODE_BINARY(<input_string>, <format>)
+ *
+ * @param input_string The string to be decoded.
+ * @param format       The encoding format (HEX, BASE64, HBASE).
+ */
+@FunctionParseNode.BuiltInFunction(name = DecodeBinaryFunction.NAME, args = {
+        @FunctionParseNode.Argument(allowedTypes = {PVarchar.class}),
+        @FunctionParseNode.Argument(enumeration = "EncodeFormat")})
+public class DecodeBinaryFunction extends DecodeFunction {
```

Comment:
This leaves us with a less capable DECODE() function, and the more capable DECODE_BINARY() function.

Since we are only extending the functionality, we could use the exact same implementation, adding more features to DECODE() is not a problem.

The only reason we are adding new functions at all is the "interesting" existing ENCODE() function.

Reply:
Technically, we do not even need the new DECODE_BINARY function, we are only adding it so that the encoding/decoding functions have intuitive names.